projects/plugins/wpcomsh/bin/i18n/generate-pot.sh

```diff
@@ -14,6 +14,8 @@ find . -name '*.php' -and ! -path './build/*' -and ! -path './custom-colors/*' \
     -or -path './vendor/automattic/jetpack-config/*' \
     -or -path './vendor/automattic/jetpack-post-list/*' \
  \) \
+ -and ! -path './vendor/*/vendor/*' \
+ -and ! -path './vendor/*/wordpress/*' \
```

Comment:
While all this seems to work, it spits out a bunch of warnings when I try it:
```
find: warning: you have specified the global option -maxdepth after the argument -name, but global options are not positional, i.e., -maxdepth affects tests specified before it as well as those specified after it.  Please specify global options before other arguments.
find: File system loop detected; ‘./vendor/automattic/jetpack-status/vendor/automattic/jetpack-plans/vendor/automattic/jetpack-status’ is part of the same file system loop as ‘./vendor/automattic/jetpack-status’.
find: File system loop detected; ‘./vendor/automattic/jetpack-status/vendor/automattic/jetpack-redirect/vendor/automattic/jetpack-status’ is part of the same file system loop as ‘./vendor/automattic/jetpack-status’.
find: File system loop detected; ‘./vendor/automattic/jetpack-status/vendor/automattic/jetpack-connection/vendor/automattic/jetpack-status’ is part of the same file system loop as ‘./vendor/automattic/jetpack-status’.
find: File system loop detected; ‘./vendor/automattic/jetpack-config/vendor/automattic/jetpack-search/vendor/automattic/jetpack-config’ is part of the same file system loop as ‘./vendor/automattic/jetpack-config’.
find: File system loop detected; ‘./vendor/automattic/jetpack-config/vendor/automattic/jetpack-sync/vendor/automattic/jetpack-config’ is part of the same file system loop as ‘./vendor/automattic/jetpack-config’.
find: File system loop detected; ‘./vendor/automattic/jetpack-config/vendor/automattic/jetpack-my-jetpack/vendor/automattic/jetpack-config’ is part of the same file system loop as ‘./vendor/automattic/jetpack-config’.
find: File system loop detected; ‘./vendor/automattic/jetpack-config/vendor/automattic/jetpack-wordads/vendor/automattic/jetpack-config’ is part of the same file system loop as ‘./vendor/automattic/jetpack-config’.
find: File system loop detected; ‘./vendor/automattic/jetpack-config/vendor/automattic/jetpack-publicize/vendor/automattic/jetpack-config’ is part of the same file system loop as ‘./vendor/automattic/jetpack-config’.
find: File system loop detected; ‘./vendor/automattic/jetpack-sync/vendor/automattic/jetpack-config/vendor/automattic/jetpack-sync’ is part of the same file system loop as ‘./vendor/automattic/jetpack-sync’.
find: File system loop detected; ‘./vendor/automattic/jetpack-sync/vendor/automattic/jetpack-search/vendor/automattic/jetpack-sync’ is part of the same file system loop as ‘./vendor/automattic/jetpack-sync’.
find: File system loop detected; ‘./vendor/automattic/jetpack-sync/vendor/automattic/jetpack-my-jetpack/vendor/automattic/jetpack-sync’ is part of the same file system loop as ‘./vendor/automattic/jetpack-sync’.
find: File system loop detected; ‘./vendor/automattic/jetpack-sync/vendor/automattic/jetpack-connection/vendor/automattic/jetpack-sync’ is part of the same file system loop as ‘./vendor/automattic/jetpack-sync’.
find: File system loop detected; ‘./vendor/automattic/jetpack-plans/vendor/automattic/jetpack-status/vendor/automattic/jetpack-plans’ is part of the same file system loop as ‘./vendor/automattic/jetpack-plans’.
find: File system loop detected; ‘./vendor/automattic/jetpack-redirect/vendor/automattic/jetpack-status/vendor/automattic/jetpack-redirect’ is part of the same file system loop as ‘./vendor/automattic/jetpack-redirect’.
find: File system loop detected; ‘./vendor/automattic/jetpack-connection/vendor/automattic/jetpack-status/vendor/automattic/jetpack-connection’ is part of the same file system loop as ‘./vendor/automattic/jetpack-connection’.
find: File system loop detected; ‘./vendor/automattic/jetpack-connection/vendor/automattic/jetpack-sync/vendor/automattic/jetpack-connection’ is part of the same file system loop as ‘./vendor/automattic/jetpack-connection’.
find: File system loop detected; ‘./vendor/automattic/jetpack-connection/vendor/automattic/jetpack-licensing/vendor/automattic/jetpack-connection’ is part of the same file system loop as ‘./vendor/automattic/jetpack-connection’.
```
I'd recommend using `-prune` to skip processing of the nested `vendor` and `wordpress` (and also `node_modules` and `jetpack_vendor`), perhaps something like this
```bash
find -L . \
  \( \
    -path './vendor/*/vendor' \
    -or -path './vendor/*/wordpress' \
    -or -path './vendor/*/node_modules' \
    -or -path './vendor/*/jetpack_vendor' \
  \) \
  -prune \
  -or \
  -name '*.php' -and ! -path './build/*' -and ! -path './custom-colors/*' \
  -and \( \
    ! -path './vendor/*' \
    -or -path './vendor/automattic/jetpack-mu-wpcom/*' \
    -or -path './vendor/automattic/at-pressable-podcasting/*' \
    -or -path './vendor/automattic/custom-fonts-typekit/*' \
    -or -path './vendor/automattic/custom-fonts/*' \
    -or -path './vendor/automattic/jetpack-assets/*' \
    -or -path './vendor/automattic/jetpack-config/*' \
    -or -path './vendor/automattic/jetpack-post-list/*' \
 \) \
 -print
```

Reply:
Thanks for the suggestion! I've updated the PR.

Although I wasn't seeing the warnings locally, prior to adding `-maxdepth` in my approach, the `find` command was taking long time to execute, which I assume was indeed a file system loop.

Reply:
Should this approach be used to automatically update translations on `jetpack-mu-wpcom`, or is the approach used in [Calypso-Bridge](https://github.com/Automattic/wc-calypso-bridge/pull/1472) more appropriate?

Reply:
Ideally WordPress Core's mechanism for loading translation bundles should be used instead of bundling files with the plugin. How to get it to load the bundles for wpcomsh from translate.wordpress.com (rather than translate.wordpress.org) I don't know.

If files have to be bundled for some reason, I don't think it matters a whole lot whether `wp i18n make-pot` or `xgettext` is used, other than that the former needs less special configuration.

P.S. For GlotPress to pick up the strings needing translation from packages, someone would probably need to have wpcomsh start using [automattic/jetpack-composer-plugin](https://packagist.org/packages/automattic/jetpack-composer-plugin) to move the relevant packages out of `vendor/`, like we do for most other monorepo plugins.